benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,647 @@
+            for col, offset in col_offsets.items():
+                df_replicated[col] += offset * r
+        
+            df = cudf.concat([df, df_replicated], ignore_index=True)
+    
+    return df
+
+
+@get_allocation_counts_dask_lazy(return_allocations=True, logging=True)
+def sample_graph(G, label_df, output_path,seed=42, batch_size=500, seeds_per_call=200000, batches_per_partition=100, fanout=[5, 5, 5]):
+    cupy.random.seed(seed)
+
+    sampler = BulkSampler(
+        batch_size=batch_size,
+        output_path=output_path,
+        graph=G,
+        fanout_vals=fanout,
+        with_replacement=False,
+        random_state=seed,
+        seeds_per_call=seeds_per_call,
+        batches_per_partition=batches_per_partition,
+        log_level = 'INFO'
+    )
+
+    n_workers = len(default_client().scheduler_info()['workers'])
+
+    meta = cudf.DataFrame({
+        'node': cudf.Series(dtype='int64'),
+        'batch': cudf.Series(dtype='int32')
+    })
+
+    
+    batch_df = label_df.map_partitions(_make_batch_ids, batch_size, n_workers, meta=meta).persist()
+    del label_df
+    print('created batches')
+    
+
+    start_time = perf_counter()
+    sampler.add_batches(batch_df, start_col_name='node', batch_col_name='batch')
+    sampler.flush()
+    end_time = perf_counter()
+    print('flushed all batches')
+    return (end_time - start_time)
+
+
+def assign_offsets_pyg(node_counts: Dict[str, int], replication_factor:int=1):
+    # cuGraph-PyG assigns offsets based on lexicographic order
+    node_offsets = {}
+    node_offsets_replicated = {}
+    count = 0
+    count_replicated = 0
+    for node_type in sorted(node_counts.keys()):
+        node_offsets[node_type] = count
+        node_offsets_replicated[node_type] = count_replicated
+
+        count += node_counts[node_type]
+        count_replicated += node_counts[node_type] * replication_factor
+    
+    return node_offsets, node_offsets_replicated, count_replicated
+
+def generate_rmat_dataset(dataset, seed=62, labeled_percentage=0.01, num_labels=256, reverse_edges=False):
+    """
+    Generates an rmat dataset.  Currently does not support heterogeneous datasets.
+
+    Parameters
+    ----------
+    dataset: The specifier of the rmat dataset (i.e. rmat_20_16)
+    seed: The seed to use for random number generation
+    num_labels: The number of classes for the labeled nodes
+    reverse_edges: Whether to reverse the edges in the edgelist (should be True for DGL, False, for PyG)
+    """
+
+    dataset = dataset.split('_')
+    scale = int(dataset[1])
+    edgefactor = int(dataset[2])
+
+    dask_edgelist_df = generate_edgelist_rmat(
+        scale=scale, edgefactor=edgefactor, seed=seed, unweighted=True, mg=True,
+    )
+    dask_edgelist_df = dask_edgelist_df.astype("int64")
+    dask_edgelist_df = dask_edgelist_df.reset_index(drop=True)
+
+
+    dask_edgelist_df = renumber_ddf(dask_edgelist_df).persist()
+    dask_edgelist_df = symmetrize_ddf(dask_edgelist_df).persist()
```

Comment:
Persist causes a big memory overhead, please avoid it. 
```suggestion
    dask_edgelist_df = renumber_ddf(dask_edgelist_df)
    dask_edgelist_df = symmetrize_ddf(dask_edgelist_df)
```

Reply:
Persistence is now disabled unless enabled by choice.